### ISSUE TITLE
will it work for multivariate time series prediction  : different continues or/and discrete/category observation

### ISSUE THREAD

**Issue by Sandy4321** [robalecarova/RcppHMM#4]
great code thanks
may you clarify : 
will it work for multivariate time series prediction   both regression and classification

each row means the time series, and columns represent different continues or/and discrete/category observation.
 
1 
  where all values are continues values 
         weight    height     age   target 
time 1|   56        160    34        1.2
time 2|   77        170    54        3.5
time 3|   87        167     43       0.7
time 4|   55        198    72        0.5
time 5|   88        176     32       2.3
etc

2
or  even will it work for multivariate time series where values are mixture of continues  and categorical values
for example 2 dimensions have continues values and 3 dimensions are categorical values

			color        weight     gender  height  age      target 
time 1|    black       56             m   160    34         yes
time 2|    white       77             f   170    54         no
time 3|   yellow      87             m   167    43          yes
time 4|    white       55             m   198    72         no
time 5|   white       88              f   176    32         yes

etc

**Comment by robalecarova** [comment]
Hi,

Right know the code only works for classification and it does not accept a mix of continuous and categorical. It just accept one or the other.

For how to use it, columns are the time steps, the rows are the features. And if you have different day observations, it can use a 3D matrix where each slice is a day observation.

A quick example of multivariate continuous data is the next:

Values for a hidden Markov model with continuous observations                          
Number of hidden states = 2
Multivariate gaussian mixture model
Observed vector with dimensionality of 3

# Initialization
N = c("X1","X2")
M <- 3

Sigma <- array(0, dim =c(M,M,length(N)))
Sigma[,,1] <- matrix(c(1.0,0.8,0.8,
                       0.8,1.0,0.8,
                       0.8,0.8,1.0), ncol = M,  
                     byrow = TRUE)
Sigma[,,2] <- matrix(c(1.0,0.4,0.6,
                       0.4,1.0,0.8,
                       0.6,0.8,1.0), ncol = M,
                     byrow = TRUE)
Mu <- matrix(c(0, 5, 
               10, 0, 
               5, 10), 
             nrow = M, 
             byrow = TRUE)

A <- matrix(c(0.6,0.4,
              0.3, 0.7), 
            ncol = length(N),
            byrow = TRUE)
Pi <- c(0.5, 0.5)

HMM.cont.multi <- verifyModel(list( "Model" = "GHMM",
                                    "StateNames" = N,
                                    "A" = A, 
                                    "Mu" = Mu, 
                                    "Sigma" = Sigma, 
                                    "Pi" = Pi))

# Data simulation
set.seed(100)
length <- 100
seqs <- 50

# Multiple sequences to be evaluated
observationSequences<- array(0, dim = c(M, length, seqs) )
for(i in 1:seqs){
  Y <- generateObservations(HMM.cont.multi , length)$Y
  observationSequences[,,i] <- Y
}

dim(observationSequences)

# New model random initialization - Model to be trained from observations
set.seed(1000)
newModel <- initGHMM(2, M) 

newModel <- learnEM(newModel,
                    observationSequences,
                    iter= 50, 
                    delta = 1E-5,
                    print = FALSE)


print(newModel)

**Comment by Sandy4321** [comment]
And if you have different day observations, it can use a 3D matrix where each slice is a day observation.

looks really great, like only you in whole github have it

can you kindly share code example for such a data prediction code?

**Comment by Sandy4321** [comment]
better for categorical data , pls

**Comment by robalecarova** [comment]
Sure, there are more examples in the newest version. You can find it in the CRAN repository.  Here is the manual and each functions includes an example:

https://cran.r-project.org/web/packages/RcppHMM/RcppHMM.pdf

The most of the examples include categorical data, discrete/counting data and continuous data.

**Comment by Sandy4321** [comment]
great only I see here
Description
This package can model observations based on hidden Markov models. The observations can be
considered to be emitted by a **multinomial distribution**,

but categorical variables they are not described by "multinomial distribution"

for example
		color             gender  city      target 
time 1| black  m abc yes
time 2| white  f xdr no
time 3| yellow  m abc yes
time 4| white  m nmk no
time 5| white  f abc ye


for example we see from this link
https://scikit-learn.org/stable/modules/classes.html#module-sklearn.naive_bayes

there are separate data models
naive_bayes.MultinomialNB
The multinomial distribution normally requires integer feature counts. 
in example above data is not _counts_
and
naive_bayes.CategoricalNB
 Naive Bayes classifier for categorical features

The categorical Naive Bayes classifier is suitable for classification with discrete features that are categorically distributed. The categories of each feature are drawn from a categorical distribution.

**Comment by robalecarova** [comment]
Yeah, the terminology may not be the best. By discrete I meant: variables that follow a Poisson Distribution, like the number of people that goes to the supermarket each day (counts).  And by categorical data, I meant data that can be expressed via labels, like in your example the 'color' would be multinomial distribution.

So yeah, my bad. Wrong terminology.

**Comment by Sandy4321** [comment]
great
but still is not clear 
1

you wrote
"The observed values can be modeled by a multinomial distribution for categorical/labeled emissions, a"

what you mean under "multinomial "?
2

to make it clear
can you share code example for 
this example: 4 categorical features and classification task (yes and no ) 

time 1| black     abc      m     qq    yes
time 2| white     abc      f       vv    no
time 3| yellow   qwer   m     vv    yes
time 4| white    qwer    f      qq    no
time 5| white    qwer    f      qq   yes

etc

Thank you very much in advance
